src/profiles/idw/custom_beneficiation.py

```diff
@@ -62,8 +62,11 @@ def beneficiate(self, yaml_path: Path) -> IngestionManifest:
                 if isinstance(obj, RawDataset):
                     ingestible_dclasses.add_dataset(obj)
                 if isinstance(obj, RawDatafile):
-                    df_path = yaml_path.parent.joinpath(obj.directory)
-                    df_dir = df_path / obj.filename
+                    if obj.directory is not None:
+                        df_path = yaml_path.parent.joinpath(obj.directory)
+                        df_dir = df_path / obj.filename
+                    else:
+                        df_dir = yaml_path.parent / obj.filename
```

Comment:
```suggestion
                    df_dir = yaml_path.parent / obj.filepath
```

Would it be equivalent just to write this?

Also on a sidenote, wondering if the name `df_dir` is a bit misleading since it represents a file rather than a directory. I know the name was already in use before your changes, but maybe it's worth changing it now while you're working on this code. Maybe it could be called `df_path`?

Reply:
Done